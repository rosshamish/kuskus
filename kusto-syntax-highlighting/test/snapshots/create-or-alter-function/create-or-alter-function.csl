.create-or-alter function with (folder='foo/bar', docstring='foobar')
fooBar(a:int, b:string) {
    let b = c;
    T
    | where foo == 1234 and bar and baz !in beer
    | join kind=innerunique hint.strategy=broadcast (
        A | where b=2
    ) on $left.a == $right.b
    | parse kind=regex textField with "Abc=" abc, "Xyz=" xyz
    | where acos(b) == acosb
    | evaluate sql_request()
    | sort by a desc null last;
}