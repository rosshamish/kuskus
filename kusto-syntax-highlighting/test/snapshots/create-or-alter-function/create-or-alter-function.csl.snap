>.create-or-alter function with (folder='foo/bar', docstring='foobar')
#^^^^^^^^^^^^^^^^ source.kusto keyword.control.kusto
#                ^ source.kusto
#                 ^^^^^^^^ source.kusto storage.type.kusto
#                         ^ source.kusto
#                          ^^^^ source.kusto keyword.other.operator.kusto
#                              ^^ source.kusto
#                                ^^^^^^ source.kusto keyword.other.operator.kusto
#                                      ^ source.kusto
#                                       ^ source.kusto string.quoted.single.kusto punctuation.definition.string.kusto
#                                        ^^^^^^^ source.kusto string.quoted.single.kusto
#                                               ^ source.kusto string.quoted.single.kusto punctuation.definition.string.kusto
#                                                ^^ source.kusto
#                                                  ^^^^^^^^^ source.kusto keyword.other.operator.kusto
#                                                           ^ source.kusto
#                                                            ^ source.kusto string.quoted.single.kusto punctuation.definition.string.kusto
#                                                             ^^^^^^ source.kusto string.quoted.single.kusto
#                                                                   ^ source.kusto string.quoted.single.kusto punctuation.definition.string.kusto
#                                                                    ^^ source.kusto
>fooBar(a:int, b:string) {
#^^^^^^^^^ source.kusto
#         ^^^ source.kusto storage.type.kusto
#            ^^^^ source.kusto
#                ^^^^^^ source.kusto storage.type.kusto
#                      ^^^^ source.kusto
>    let b = c;
#^^^^ source.kusto
#    ^^^ source.kusto keyword.control.kusto
#       ^^^^^^^^ source.kusto
>    T
#^^^^^^ source.kusto
>    | where foo == 1234 and bar and baz !in beer
#^^^^^^ source.kusto
#      ^^^^^ source.kusto keyword.other.query.kusto
#           ^^^^^^^^ source.kusto
#                   ^^^^ source.kusto constant.numeric.kusto
#                       ^ source.kusto
#                        ^^^ source.kusto keyword.other.operator.kusto
#                           ^^^^^ source.kusto
#                                ^^^ source.kusto keyword.other.operator.kusto
#                                   ^^^^^ source.kusto
#                                        ^^^ source.kusto keyword.other.operator.kusto
#                                           ^^^^^^ source.kusto
>    | join kind=innerunique hint.strategy=broadcast (
#^^^^^^ source.kusto
#      ^^^^ source.kusto keyword.other.query.kusto
#          ^ source.kusto
#           ^^^^ source.kusto keyword.other.operator.kusto
#               ^ source.kusto
#                ^^^^^^^^^^^ source.kusto keyword.other.kusto
#                           ^ source.kusto
#                            ^^^^^^^^^^^^^ source.kusto keyword.other.operator.kusto
#                                         ^ source.kusto
#                                          ^^^^^^^^^ source.kusto keyword.other.kusto
#                                                   ^^^ source.kusto
>        A | where b=2
#^^^^^^^^^^^^ source.kusto
#            ^^^^^ source.kusto keyword.other.query.kusto
#                 ^^^ source.kusto
#                    ^ source.kusto constant.numeric.kusto
>    ) on $left.a == $right.b
#^^^^^^ source.kusto
#      ^^ source.kusto keyword.other.operator.kusto
#        ^ source.kusto
#         ^^^^^ source.kusto keyword.other.kusto
#              ^^^^^^ source.kusto
#                    ^^^^^^ source.kusto keyword.other.kusto
#                          ^^^ source.kusto
>    | parse kind=regex textField with "Abc=" abc, "Xyz=" xyz
#^^^^^^ source.kusto
#      ^^^^^ source.kusto keyword.other.query.kusto
#           ^ source.kusto
#            ^^^^ source.kusto keyword.other.operator.kusto
#                ^ source.kusto
#                 ^^^^^ source.kusto keyword.other.operator.kusto
#                      ^^^^^^^^^^^ source.kusto
#                                 ^^^^ source.kusto keyword.other.operator.kusto
#                                     ^ source.kusto
#                                      ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
#                                       ^^^^ source.kusto string.quoted.double.kusto
#                                           ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
#                                            ^^^^^^ source.kusto
#                                                  ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
#                                                   ^^^^ source.kusto string.quoted.double.kusto
#                                                       ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
#                                                        ^^^^^ source.kusto
>    | where acos(b) == acosb
#^^^^^^ source.kusto
#      ^^^^^ source.kusto keyword.other.query.kusto
#           ^ source.kusto
#            ^^^^ source.kusto support.function.kusto
#                ^^^^^^^^^^^^^ source.kusto
>    | evaluate sql_request()
#^^^^^^ source.kusto
#      ^^^^^^^^ source.kusto keyword.other.query.kusto
#              ^ source.kusto
#               ^^^^^^^^^^^ source.kusto support.function.kusto
#                          ^^^ source.kusto
>    | sort by a desc null last;
#^^^^^^ source.kusto
#      ^^^^ source.kusto meta.query.sort.kusto keyword.other.query.kusto
#          ^ source.kusto meta.query.sort.kusto
#           ^^ source.kusto meta.query.sort.kusto keyword.other.operator.kusto
#             ^^^ source.kusto
#                ^^^^ source.kusto keyword.other.kusto
#                    ^ source.kusto
#                     ^^^^ source.kusto constant.language.kusto
#                         ^^^^^^^ source.kusto
>}
#^^ source.kusto